src/VirtoCommerce.Platform.Security/Repositories/SecurityRepository.cs

```diff
@@ -1,18 +1,37 @@
 using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.Security;
 using VirtoCommerce.Platform.Data.Infrastructure;
 using VirtoCommerce.Platform.Security.Model;
 
 namespace VirtoCommerce.Platform.Security.Repositories
 {
     public class SecurityRepository : DbContextRepositoryBase<SecurityDbContext>, ISecurityRepository
     {
-        public SecurityRepository(SecurityDbContext dbContext)
+        private readonly PasswordOptionsExtended _passwordOptions;
```

Comment:
I think you should not inject PasswordOptionsExtended into this repository, please make the GetUserPasswordsHistoryAsync method more decoupled from the configuration options, better to receive the history count as an argument